### PR TITLE
Added a functionality to display descriptions of key terms when they are hovered over

### PR DESCRIPTION
**Pull request by smob123**
added a functionality to display descriptions of key terms when they are hovered over as described on the trello board:

https://trello.com/c/pt4dx8s0/37-show-a-description-of-key-terms-when-they-are-hovered-over
___
NB - by making a pull request you confirm agreement to the [Contributors Agreement](https://github.com/openlawnz/openlawnz-web/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by smob123** [comment]
[Show a description of 'key terms' when they are hovered over](https://trello.com/c/pt4dx8s0/37-show-a-description-of-key-terms-when-they-are-hovered-over)

**Comment by BenRauzi** [comment]
Looks really good, just a few things right now:

- When you join the text back together you aren't adding a space between the words again, so it's just joining as one long 'word' with no spaces. (Check out the other two microsite examples and you'll see)
- Can you make up so that the icon lines up with the top and bottom of the text. (Right now the centre of the circle is at the centre of the text - this means that it sticks out below the bottom of the line a bit.
- Would you be able to add some rounding to the edge of the corners to line up to the designs

Just a few small changes. I'll get Will to take a look over after.

File: src/templates/microsite.js [hunk 1]
```diff
@@ -1,48 +1,71 @@
...
-                />
-              </div>
-            )
-          })
+          splitText[index] = splitText[index].replace(word, newHtml);
         }
+      })
+    });
+
+    return splitText.join('');
```
Comment (BenRauzi) [inline src/templates/microsite.js:29]: See here - no spaces between words/divs

**Comment by BenRauzi** [comment]
My only thought may be that people may want to copy content from the 'Key Term' box, but not sure if that is relevant. See what Will thinks.

File: src/scss/Microsite.scss [hunk 2]
```diff
@@ -1,28 +1,77 @@
+$openLawA11yBrown: #a36633;
```
Comment (williamparry) [inline src/scss/Microsite.scss:1]: Should this be in a common file?

File: src/templates/microsite.js [hunk 3]
```diff
@@ -1,48 +1,71 @@
...
- 
+const Microsite = ({ pageContext }) => {
+  function highlightKeyTerms(keyTerms, html) {
+    const splitText = html.split(' ');
 
+    splitText.forEach((word, index) => {
+      keyTerms.forEach(term => {
+        if (word.includes(term.title)) {
+          const newHtml = `
+          <span class='key-term'>
```
Comment (williamparry) [inline src/templates/microsite.js:15]: Could this be more semantically rich?

File: src/scss/Microsite.scss (hunk 2 above)
Comment (BenRauzi) [reply src/scss/Microsite.scss:1]: ^ This already exists in [app.scss](https://github.com/openlawnz/openlawnz-web/blob/master/src/scss/App.scss) so shouldn't be needed here.

File: src/templates/microsite.js [hunk 4]
```diff
@@ -1,48 +1,71 @@
...
+            <h2>{pageContext.title} - {pageContext.section.title}</h2>
+            <div className="microsite-section" name={pageContext.section.title}>
+              {
+                pageContext.section.paragraphs.map(({ title, content_html }, idx) => {
+                  return (
+                    <div key={idx} name={title.replace(/\s/g, '-').toLowerCase()}>
+                      <h4>{title}</h4>
+                      <div
+                        className="microsite-paragraph"
+                        dangerouslySetInnerHTML={{ __html: highlightKeyTerms(pageContext.keyTerms, content_html) }}
```
Comment (williamparry) [inline src/templates/microsite.js:47]: Can we make sure to dompurify this?

Comment (BenRauzi) [reply src/templates/microsite.js:47]: @williamparry This is actually one I flagged on another PR. Does this even need to be done using dangerouslySetInnerHTML? My thinking is that it'll probably only be raw text, in which case we could go for just text?

Comment (smob123) [reply src/templates/microsite.js:47]: wouldn't changing this break the Consumer Credit and Tenancies pages?

File: src/scss/Microsite.scss (hunk 2 above)
Comment (smob123) [reply src/scss/Microsite.scss:1]: Tried to use the colour without re-declarings the variables but got this error "Undefined variable: "$openLawGold"". Do you guys have any suggestions?

**Comment by BenRauzi** [comment]
@williamparry Could you take another look over this for @smob123 when you have a chance? I'm happy to handle the merge conflicts after.

**Pull request closed without merging**